### ISSUE TITLE
Error when constructing `Normal` distribution from length-1 lists

### ISSUE THREAD

**Issue by nathanaelbosch** [probabilistic-numerics/probnum#116]
```python
RandomVariable(distribution=Normal([1.],[2.]))
```
returns:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-38-4ae9c921b88f> in <module>
----> 1 RandomVariable(distribution=Normal([1.],[[2.]]))

~/PhD-Projects/probnum/src/probnum/prob/distributions/normal.py in __new__(cls, mean, cov, random_state)
     75         if cls is Normal:
     76             if _both_are_univariate(mean, cov):
---> 77                 return _UnivariateNormal(mean, cov, random_state)
     78             elif _both_are_multivariate(mean, cov):
     79                 return _MultivariateNormal(mean, cov, random_state)

~/PhD-Projects/probnum/src/probnum/prob/distributions/normal.py in __init__(self, mean, cov, random_state)
    353 
    354     def __init__(self, mean=0.0, cov=1.0, random_state=None):
--> 355         super().__init__(mean=float(mean), cov=float(cov), random_state=random_state)
    356 
    357     def var(self):

TypeError: float() argument must be a string or a number, not 'list'
```

It seems `_both_are_univariate` returns true for `np.arrays` that can be considered to correspond to scalars, but `_UnivariateNormal` is actually only able to handle scalars. 
Personally, I would probably prefer only using "Univariate" if scalars are passed, and using "Multivariate" if arrays or arry-like's are passed, even if they only contain a single dimension.

From looking into the code, it also seems like `Normal` is actually not able to handle "array-like" types, even though the documentation explicitly states it.

**Comment by pnkraemer** [comment]
Seems to me that at least the last sentence is related to  #89.

**Comment by JonathanWenger** [comment]
This is also linked to #94 .

**Comment by marvinpfoertner** [comment]
Should be resolved with #135.
